BDArmory/GameModes/ModuleSpaceFriction.cs

```diff
@@ -196,19 +196,33 @@ public void FixedUpdate()
 
                         Vector3d grav = FlightGlobals.getGeeForceAtPosition(vessel.CoM);
                         var vesselMass = part.vessel.GetTotalMass();
-                        using (var repulsor = repulsors.GetEnumerator())
-                            while (repulsor.MoveNext())
-                            {
-                                if (repulsor.Current == null) continue;
-                                float pointAltitude = BodyUtils.GetRadarAltitudeAtPos(repulsor.Current.transform.position);
-                                if (pointAltitude <= 0 || pointAltitude > 2f * targetAlt) continue;
-                                var factor = Mathf.Clamp(Mathf.Exp(BDArmorySettings.SF_REPULSOR_STRENGTH * (targetAlt - pointAltitude) / targetAlt - (float)vessel.verticalSpeed / targetAlt), 0f, 5f * BDArmorySettings.SF_REPULSOR_STRENGTH); // Decaying exponential balanced at the target altitude with velocity damping.
-                                float repulsorForce = vesselMass * factor / repulsors.Count; // Spread the force between the repulsors.
-                                if (float.IsNaN(factor) || float.IsInfinity(factor)) // This should only happen if targetAlt is 0, which should never happen.
-                                    Debug.LogWarning($"[BDArmory.Spacehacks]: Repulsor Force is NaN or Infinity. TargetAlt: {targetAlt}, point Alt: {pointAltitude}, VesselMass: {vesselMass}");
-                                else
-                                    repulsor.Current.part.Rigidbody.AddForce(-grav * repulsorForce, ForceMode.Force);
-                            }
+                        if (RepulsorOverride)
+                        {
+                            float pointAltitude = BodyUtils.GetRadarAltitudeAtPos(part.transform.position);
+                            if (pointAltitude <= 0 || pointAltitude > 2f * targetAlt) return;
+                            var factor = Mathf.Clamp(Mathf.Exp(BDArmorySettings.SF_REPULSOR_STRENGTH * (targetAlt - pointAltitude) / targetAlt - (float)vessel.verticalSpeed / targetAlt), 0f, 5f * BDArmorySettings.SF_REPULSOR_STRENGTH); // Decaying exponential balanced at the target altitude with velocity damping.
+                            float repulsorForce = vesselMass * factor / repulsors.Count; // Spread the force between the repulsors.
+                            if (float.IsNaN(factor) || float.IsInfinity(factor)) // This should only happen if targetAlt is 0, which should never happen.
+                                Debug.LogWarning($"[BDArmory.Spacehacks]: Repulsor Force is NaN or Infinity. TargetAlt: {targetAlt}, point Alt: {pointAltitude}, VesselMass: {vesselMass}");
+                            else
+                                part.Rigidbody.AddForce(-grav * repulsorForce, ForceMode.Force);
```

Comment:
The part that this is being applied to is the root part of the vessel, yes? Won't that make most vessels that use this spin and dangle from the root part if the root part isn't close to the CoM?

Reply:
no, it'd be whatever part(s) has a 
```
MODULE
{
 name = ModuleSpaceFriction
 RepulsorOverride = true.
 }
``` in it's .cfg or added via MM patch.

Reply:
OK, but there's a number of other places in the code where `ModuleSpaceFriction` is being added to the root part if it's not there (SpawnUtils.cs, VesselSpawnerBase.cs and MissileFire.cs (this one isn't guarded by `SPACE_HACKS`)), which likely interefere with how this is intended to work.

Also, the check for a duplicate `ModuleSpaceFriction` on the root part in `Start` probably shouldn't run for parts other than the root part.

Reply:
Yes, I am aware, I tested with Repulsor Mode both on and off. The one added to root via the SpawnUtilities for `SPACEHACKS` use doesn't have the `RepulsorOverride` bool, so it would only activate if `Repulsor Mode` is activated, and the `if (RepulsorOverride)... else` would keep the repulsor parts from adding a lifting effect to landing gear (and vice-versa), if for whatever reason both repulsor parts are present and Repulsor Mode is activated. 
The only place you'd get a doubled up repulsor effect is if there are external repulsor parts and landing gear on the craft, and Repulsor Mode is activated all at once, which would make the craft more bouncy, but still respect the target alt